### ISSUE TITLE
Please add formatted file name information as meta data if none found

### ISSUE THREAD

**Issue by bkndc** [CodyBerenson/PGMA-Modernized#43]
If the movie/scene etc can not be found on any of the sites, due to any number of reasons.  Can the formatted file name information be plugged into and used as the metadata as part of the Local Media Resources agent (temporarily at least)?  I have over 8,000 movies in my porn collection.  I have painstakingly spent hours going through and renaming all the files in the standard of "(Studio) - Movie Title (YYYY)."   I have created a new NAS Plex Server with brand new drives and decided that I am going to place all movies in one Movies Directory/Library.   I have downloaded and installed the PGMA-Modernized Bundles.  And configured the Plugin/Agent with Local Media Resources as last in the list.  And using Gay Adult as the Library type and correct scanner.  I started adding the movies to the library/Directory before I pointed Plex to it.  

I have only imported about 100 movies/scenes so far.  For those movies/scenes that cannot be found can the (Studio) be added to the "Studio" Tag?  Can the "Movie Title" be added to the "Movie Title" "Sort Title" and the (YYYY) based on the file name?  

If this information were plugged into the metadata I could at least use some of the filterings and sorting features of Plex without having all the metadata from a web site.  And all the hard work of renaming the movies will be worth something.

There should be an easy way to see if movies didn't get scrapped or there was not a match in bulk.  Maybe there is and I am not seeing it.  

![ScreentPrint1](https://user-images.githubusercontent.com/67534775/91646423-1118a380-ea1d-11ea-8fbf-2008551ec32c.PNG)
![screenprint2](https://user-images.githubusercontent.com/67534775/91646424-1675ee00-ea1d-11ea-992f-1f9ec61308cc.PNG)
![screentprint3](https://user-images.githubusercontent.com/67534775/91646427-1970de80-ea1d-11ea-9e95-cfd1c417f8ab.PNG)
![screenprint4](https://user-images.githubusercontent.com/67534775/91646428-1e359280-ea1d-11ea-87c6-b8b2cf602999.PNG)
![ScreenPrint5](https://user-images.githubusercontent.com/67534775/91646430-21308300-ea1d-11ea-9813-75b50a85d0ad.PNG)
![screentprint6](https://user-images.githubusercontent.com/67534775/91646431-2392dd00-ea1d-11ea-875c-f8e3a10b3b31.PNG)

Thank you again for your hard work.

**Comment by CodyBerenson** [comment]
@bkndc I am not sure how or why, but this issue only showed up today.  This is so odd, as other issues have been opened and resolved since. Anyway, GitHub quirks aside I didnt want you to think you were being ignored. Once I start my day I’ll take a look at  what you‘re asking.

**Comment by CodyBerenson** [comment]
@bkndc 

I'm not sure if in the time between you created this issue (#43) and the Issue created yesterday #48 if you've made progress on naming your files.  I'm going to assume not, and see if I can help.  I think I see the misunderstanding.

Let's take the example you give above pictured from your screen shots:  (Broke Straight Boys) - John Henry Xavier Ryan (2018).mp4.   Although from a purely syntax perspective you may believe this is named correctly, it is not; here's why:

Step 1.  You must find a match for scene on one of the five scene sites that we have built agents for (technically four since Fagalicious is currently not working):  WayBig.com, QueerClick.com, Fagalicious.com (not currently working), IAFD.com, or BestExclusivePorn.com.

Step 2.  I found that scene on 4 of the 5 above (it doesn't seem to be BestExclusivePorn.com).

Step 3.  Choose one, depending on your preferred site.  Most folks seem to like WayBig.com, so let's go with that.  Here is the address for the scene:  
https://www.waybig.com/blog/?s=john+henry+xavier+ryan

and a partial screen shot:
![image](https://user-images.githubusercontent.com/40373801/94115587-f1a44900-fe17-11ea-8530-d041ec50d24d.png)

Step 4.  Name the file using the title from the site.  Again, targeting a Plex match using the WayBig (IAFD) agent, properly name the file based on the title from the site:  **(Broke Straight Boys) - John Henry Fucks Xavier Ryan (Bareback) (2018).mp4**

Step 5.  Rescan the library.  If Plex doesn't correct the match (for me, Plex is hit or miss), do a "Fix Match" for this particular scene
choosing the proper Agent set:

![image](https://user-images.githubusercontent.com/40373801/94117081-ce7a9900-fe19-11ea-88b3-6e9ceb404ad5.png)

Plex will then attempt to match the scene against each of the enabled agents.  If the WayBig (IAFD) agent is enabled, it should match the new properly named filename, return metada and poster art.  Here's a Plex Screen Shot of the matched scene:

![image](https://user-images.githubusercontent.com/40373801/94116283-d71e9f80-fe18-11ea-836d-19960c7ac76d.png)

Again, in the time since you originally wrote this and today, you may have figured all of this out.  If you're still having trouble, reach out at dakota.berenson@gmail.com and we can troubleshoot.

**Comment by bkndc** [comment]
Thank you Cody, however that syntax is different than everything else.  And will throw off the process of video naming validation. Cause what I have found is that the video has to have the title and year in the metadata before plex will go get the off the internet if the title is blank and the year then the video is not processed.  Using mp3tag I am able to add the title and year based on the (studio) - title (year).  If I add another (bareback) to the file name when I go to process the files (bareback) will end up in the year field in the year field of the metadata.  I wish the (studio) has some translation to a field in the metadata ie producer or something else.  Because unless the video matches the title of movie in plex becomes “Broke in Straight Boys Carl & Sam” for both the “title and the sorting title”.   Also my collection is over 6000 videos and scenes going back through and having to look up each video to determine if it’s a scene or video will be very time consuming.  If I change the tile I have to change the title in the meta data. Change the title of the video and change the title of the directory it’s in because also the only way to get the video to match is to put the video in a directory that is the same name as the video.  So I have a directory for each title. I used to have all the videos in one big movies directory now my path looks like this \\servername\Videos\movies\(studio) - title  (year)\(studio) - title (year).ext.   it’s the only way I have been successful at getting the videos to match.

If I also use mo3tag to populate the producers field with the studio in the metadata and that field is then read by plex and the video is the automatically put into the collections field so all my videos automatically end up in a collection of studios.  I could get fancy and add other things such as genre ie condom or bareback which it seems that guys would rather watch bareback over condom videos.  Some when the see a movie with condoms they turn to another movie.

On another note I also discovered the one of the things that mp3tag will also do is optimize the atom file in the mp4 container and moves it to the beginning of the stream so all my videos load immediately when i click on them on any device there isn’t a delay or buffering while  Plex loads the entire file so it can find the atom file to determine how to play it all that info is at the beginning of the mp4 container now.

Bryan

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
